backend-rust/src/graphql_api/baker.rs

```diff
@@ -1045,6 +1076,519 @@ impl Baker {
         }
         Ok(connection)
     }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn block_commission_desc_connection(
+        config: &ApiServiceConfig,
+        pool: &PgPool,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+        open_status_filter: Option<BakerPoolOpenStatus>,
+        include_removed_filter: bool,
+    ) -> ApiResult<connection::Connection<String, Baker>> {
+        type RemovedBakerCursor = Reversed<BakerIdCursor>;
+        type Cursor = ConcatCursor<BakerFieldDescCursor, RemovedBakerCursor>;
+
+        /// Internal helper function for querying the current bakers sorted
+        /// by the pool_delegator_count in descending order.
+        async fn query_current_bakers(
+            query: ConnectionQuery<BakerFieldDescCursor>,
+            open_status_filter: Option<BakerPoolOpenStatus>,
+            connection: &mut connection::Connection<String, Baker>,
+            pool: &PgPool,
+        ) -> ApiResult<()> {
+            let mut row_stream = sqlx::query_as!(
+                CurrentBaker,
+                r#"SELECT * FROM (
+                SELECT
+                    bakers.id AS id,
+                    staked,
+                    restake_earnings,
+                    open_status as "open_status: _",
+                    metadata_url,
+                    transaction_commission,
+                    baking_commission,
+                    finalization_commission,
+                    payday_transaction_commission as "payday_transaction_commission?",
+                    payday_baking_commission as "payday_baking_commission?",
+                    payday_finalization_commission as "payday_finalization_commission?",
+                    payday_lottery_power as "payday_lottery_power?",
+                    payday_ranking_by_lottery_powers as "payday_ranking_by_lottery_powers?",
+                    (SELECT MAX(payday_ranking_by_lottery_powers) FROM bakers_payday_lottery_powers) as "payday_total_ranking_by_lottery_powers?",
```

Comment:
Does bakers_payday_lottery_powers only return a single row or why do we return MAX here?